cmd/aws-janitor/main.go

```diff
@@ -191,8 +193,19 @@ func markAndSweep(opts resources.Options, region string) error {
 	}
 
 	sweepCount = res.MarkComplete()
-	if err := res.Save(opts.Session, s3p); err != nil {
-		return errors.Wrapf(err, "Error saving %q", *path)
+
+	saveSet := true
+	if *enableS3BucketClean{
+		if !s3path.CheckBucketExist(opts.Session, opts.Region, s3p.Bucket) {
```

Comment:
It may be even better for explicitly fail - user probably will want to know that janitor kills its own state bucket, and terminating without error makes it hard to miss.